### PR TITLE
Reject invalid job names

### PR DESCRIPTION
**Pull request by yvanoers**
Job names with illegal characters are rejected with a 400 HTTP response.
Removed job name sanitation code.
Fixed a typo.

#519

### CODE REVIEW

**Comment by yvanoers** [comment]
I started out using validator.v8 as described in the gin/gonic documentation, but that led to much messy code. So then I wrote it this way, much more straight forward and less code. At least I think so.

File: dkron/api.go [hunk 1]
```diff
@@ -248,3 +256,13 @@ func (h *HTTPTransport) jobToggleHandler(c *gin.Context) {
 	c.Header("Location", c.Request.RequestURI)
 	renderJSON(c, http.StatusOK, job)
 }
+
+// isSlug determines whether the given string is a proper value to be used as
+// key for in any backend store (a "slug"). If false, the 2nd return value
```
Comment (victorcoder) [inline dkron/api.go]: We're only using one store now in v2.

```suggestion
// key for the backend store (a "slug"). If false, the 2nd return value
```

File: dkron/api_test.go [hunk 2]
```diff
@@ -134,3 +132,35 @@ func TestAPIJobCreateUpdateParentJob_NoParent(t *testing.T) {
 	// Send a shutdown request
 	a.Stop()
 }
+
+func TestAPIJobCreateUpdateValidationBadName(t *testing.T) {
+	a := setupAPITest(t)
+
+	jsonStr := []byte(`{"name": "BAD JOB NAME!", "schedule": "@every 1m", "executor": "shell", "executor_config": {"command": "date"}, "disabled": true}`)
```
Comment (victorcoder) [inline dkron/api_test.go]: Better readability in multiline

File: dkron/api_test.go [hunk 3]
```diff
@@ -134,3 +132,35 @@ func TestAPIJobCreateUpdateParentJob_NoParent(t *testing.T) {
...
+	assert.Equal(t, http.StatusBadRequest, resp.StatusCode)
+
+	// Send a shutdown request
+	a.Stop()
+}
+
+func TestAPIJobCreateUpdateValidationValidName(t *testing.T) {
+	a := setupAPITest(t)
+
+	jsonStr := []byte(`{"name": "abcdefghijklmnopqrstuvwxyz0123456789-_ßñëäïüøüáéíóýćàèìòùâêîôûæšłç", "schedule": "@every 1m", "executor": "shell", "executor_config": {"command": "date"}, "disabled": true}`)
```
Comment (victorcoder) [inline dkron/api_test.go]: Same here, better formatting

File: dkron/api.go (hunk 1 above)
Comment (yvanoers) [reply dkron/api.go]: Done.
v2 uses only one store? Which one?

Comment (yvanoers) [reply dkron/api.go]: Found it: BadgerDB.

**Comment by yvanoers** [comment]
Conflicts are now resolved and the tests updated to work with the updated code.
Note: I accidentally committed two changes (26b1c07 and 45f9b5c) that weren't supposed to be committed to my v2.x branch. Commit b398477 reverts 26b1c07, but I left the other one in because it only has text corrections.

**Pull request merged**